src/main/java/org/zalando/nakadi/util/NakadiUtils.java

```diff
@@ -0,0 +1,26 @@
+package org.zalando.nakadi.util;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class NakadiUtils {
+    public static class Diff<T> {
+        public final Set<T> added;
+        public final Set<T> removed;
+
+        Diff(final Set<T> added, final Set<T> removed) {
+            this.added = added;
+            this.removed = removed;
+        }
+    }
+
+    public static <T> Diff<T> difference(final Set<T> oldCollection, final Set<T> newCollection) {
```

Comment:
@v-stepanov In collections values may be not unique, and that is why the way it will work won't be so straightforward.